### PR TITLE
Update and restructure categorical chapter

### PR DESCRIPTION
**Pull request by tcmetzger**
Follow-up to #13408

Restructures and updates the "Categorical" chapter as discussed in https://github.com/bokeh/bokeh/pull/13408#issuecomment-1751291318 and https://github.com/bokeh/bokeh/pull/13408#issuecomment-1751419414

### CODE REVIEW

**Review by bryevdv — APPROVED** [review]

**Comment by bryevdv** [comment]
@tcmetzger please check that my conflict resolve was correct when you have a chance, otherwise ready to merge

File: docs/bokeh/source/docs/user_guide/topics/categorical.rst [hunk 1]
```diff
@@ -71,73 +92,76 @@ This will shift each bar horizontally by the corresponding offset.
...
-the absolute value of the correlation coefficient. In this way, low correlations are suppressed
-and high correlations stand out better.
+This plot displays the correlations as colored circles. The scale of the circles
+corresponds to the absolute value of the correlation coefficient.
+This way, low correlations are suppressed and high correlations stand out better.
 
-This example uses ``linear_cmap`` to map the colors of the plot
-in order to highlight the correlations between the pair of elements.
+This example uses the :class:`~bokeh.models.mappers.LinearColorMapper` to map the
+colors of the plot in order to highlight the correlations between the pair of elements.
```
Comment (bryevdv) [inline docs/bokeh/source/docs/user_guide/topics/categorical.rst:164]: I guess this is slightly out of sync, the example was updated to use `linear_cmap` and `construct_colorbar` rathern than `LinearColorMapper` directly.

Comment (tcmetzger) [reply docs/bokeh/source/docs/user_guide/topics/categorical.rst:164]: Good point - I'll also update this for the unemployment example description!

**Comment by tcmetzger** [comment]
@bryevdv I'm still working on building this locally - I'll merge once I can confirm that everything works correctly. You can also merge it at any time, of course!

**Pull request merged**